v2/googlecloud-to-elasticsearch/docs/PubSubToElasticsearch/README.md

```diff
@@ -0,0 +1,101 @@
+# PubSub to Elasticsearch Dataflow Template
+
+The [PubSubToElasticsearch](../../src/main/java/com/google/cloud/teleport/v2/elasticsearch/templates/PubSubToElasticsearch.java) pipeline
+ingests data from a PubSub subscription, optionally applies a Javascript UDF if supplied and writes the data to Elasticsearch.
+
+## Getting Started
+
+### Requirements
+* Java 8
+* Maven
+* PubSub Subscription exists
+* Elasticsearch host(s) exists and is operational
+
+### Building Template
+This is a Flex Template meaning that the pipeline code will be containerized, and the container will be
+used to launch the Dataflow pipeline.
+
+#### Building Container Image
+* Set environment variables.
+```sh
+export PROJECT=<my-project>
+export IMAGE_NAME=<my-image-name>
+export BUCKET_NAME=gs://<bucket-name>
+export TARGET_GCR_IMAGE=gcr.io/${PROJECT}/${IMAGE_NAME}
+export BASE_CONTAINER_IMAGE=gcr.io/dataflow-templates-base/java8-template-launcher-base
+export BASE_CONTAINER_IMAGE_VERSION=latest
+export TEMPLATE_MODULE=pubsub-to-elasticsearch
+export APP_ROOT=/template/googlecloud-to-elasticsearch
+export COMMAND_SPEC=${APP_ROOT}/resources/${TEMPLATE_MODULE}-command-spec.json
+export TEMPLATE_IMAGE_SPEC=${BUCKET_NAME}/images/${TEMPLATE_MODULE}-image-spec.json
+
+export TARGET_NODE_ADDRESSES=<comma-separated-list-nodes>
+export SUBSCRIPTION=<my-subscription>
+export WRITE_INDEX=<my-index>
+export WRITE_DOCUMENT_TYPE=<my-type>
+export DEADLETTER_TABLE=<my-project:my-dataset.my-deadletter-table>
+```
+
+* Build and push image to Google Container Repository
+
+```sh
+mvn clean package -Dimage=${TARGET_GCR_IMAGE} \
+                  -Dbase-container-image=${BASE_CONTAINER_IMAGE} \
+                  -Dbase-container-image.version=${BASE_CONTAINER_IMAGE_VERSION} \
+                  -Dapp-root=${APP_ROOT} \
+                  -Dcommand-spec=${COMMAND_SPEC} \
+                  -am -pl ${TEMPLATE_MODULE}
+```
+
+#### Creating Image Spec
```

Comment:
Incorrect Image Spec.  In the new format image spec is created indicating imageuri and metadata similar to readme.md files with other templates

Reply:
@prathapreddy123 
Fixed.
Also, I made inputSubscription required field.